### DISCUSSION TITLE
How to encapsulate ox-hugo dependencies in a Docker dependency image

### DISCUSSION THREAD

**Discussion by BloggerBust** [kaushalmodi/ox-hugo#482] [Show and tell]
Hi Kaushal Modi,

I posted on [how to encapsulate ox-hugo dependencies in a Docker dependency image](https://bloggerbust.ca/post/ox-hugo-dependency-image/). The dependency image allows you to compile ox-hugo markdown and docs as well as run the tests without installing any dependencies. It also allows you to run Hugo server while composing a blog without having any dependencies installed on your host environment. That is, you do not need to have hugo, go, pandoc or pandoc-citeproc or any related dependencies installed on your host. You will need emacs installed on the host with ox-hugo of course. 

Let me know what you think. Note that I have never tested this from a windows environment.

Cheers!

Trevor.

**Comment by kaushalmodi** [comment]
Hello, that looks like a well researched post. I'm just back from a vacation. 

Happy New Year!

Let me clear up some backlog, and get back to looking at your post, if not earlier, then on this weekend.

Thanks!

**Comment by BloggerBust** [comment]
Thanks Kaushal, I would appreciate any feedback you can offer. I hope that you enjoyed your vacation. Happy New Year!

Trevor.

**Comment by BloggerBust** [comment]
There is no need to keep this as an open issue. I just wanted to communicate that it existed and since I don't have your email address I just created an issue.

In all honesty my proposed approach to encapsulating dependencies using docker might be too complicated for people to bother setting up. Also, I don't even know if it works in a windows environment as I don't own a single windows system.

I am going to close this issue. If you ever do try it out then let me know what your thoughts are.

I am still interested in contributing to your project and plan to do so in the future. I am a bit preoccupied with something else at the moment. Also, I have to brush up on my elisp skills. I mainly use it for configuring Emacs, but usually only in very trivial ways.

Cheers!

**Comment by kaushalmodi** [comment]
>In all honesty my proposed approach to encapsulating dependencies using docker might be too complicated for people to bother setting up.

Actually glancing through your post made me think that this is a huge learning curve for me. I don't have docker installed, nor is it installed at work. So it's unlikely that I'll try out the steps in your post. But I do want to go through your post so that I understand what docker is about. 

> I am going to close this issue. If you ever do try it out then let me know what your thoughts are.

You don't have to close this issue :). It being open leaves a reminder that I need to go through the issue. 

> I am still interested in contributing to your project and plan to do so in the future. I am a bit preoccupied with something else at the moment. Also, I have to brush up on my elisp skills. I mainly use it for configuring Emacs, but usually only in very trivial ways.

No worries. Being involved in a couple of open source projects, I can totally see how it's not possible to be always up to speed in all the projects while also taking care of work and life. 

I personally have a bunch of feature/doc issues to take care of in ox-hugo. But I don't have time for that currently. I only fix bugs and user questions with urgency at the moment.

**Comment by BloggerBust** [comment]
If you have any questions you can email me directly or post here. My email is linked near the top of the my blog.

> I don't have docker installed, nor is it installed at work. So it's unlikely that I'll try out the steps in your post.
 
Docker is actually very easy to install, requires no configuration or setup, and uninstalls cleanly as well. You could always install Docker at home, try it out, then uninstall it afterwards. Even if you don't try it out I would love some feedback.
